services/requester/pool.go

```diff
@@ -61,37 +65,41 @@ func (t *TxPool) Send(
 		return err
 	}
 
-	// add to pool and delete after transaction is sealed or errored out
-	t.pool.Store(evmTx.Hash(), evmTx)
-	defer t.pool.Delete(evmTx.Hash())
-
-	backoff := retry.WithMaxDuration(time.Minute*1, retry.NewConstant(time.Second*1))
-	return retry.Do(ctx, backoff, func(ctx context.Context) error {
-		res, err := t.client.GetTransactionResult(ctx, flowTx.ID())
-		if err != nil {
-			return fmt.Errorf("failed to retrieve flow transaction result %s: %w", flowTx.ID(), err)
-		}
-		// retry until transaction is sealed
-		if res.Status < flow.TransactionStatusSealed {
-			return retry.RetryableError(fmt.Errorf("transaction %s not sealed", flowTx.ID()))
-		}
-
-		if res.Error != nil {
-			if err, ok := parseInvalidError(res.Error); ok {
-				return err
+	if t.config.TxStateValidation == config.TxSealValidation {
```

Comment:
My comment is not here, but for L62 where we had

```
t.txPublisher.Publish(evmTx) // publish pending transaction event
```

Do we still need to mark the tx as sealed if TxStateValidation is local-index?

Reply:
Regardless of what mechanism we use to validate submitted transactions, the `t.txPublisher.Publish(evmTx)` is used for this [filter](https://www.quicknode.com/docs/ethereum/eth_newPendingTransactionFilter). It's a polling filter that developers can use to get notified when there's a pending transaction in the network. I have an issue for reworking the way we handle pending transactions in the codebase, so that match the functionality from Geth: https://github.com/onflow/flow-evm-gateway/issues/544.